File: code-coaches/medium/missing-numbers/MissingNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class MissingNumbers
{
    static List<int> GetMissingNumbersInSequence(List<int> nums)
    {
        int minNum = nums.Min();
        int maxNum = nums.Max();
        List<int> missingNums = new List<int>();

        for (int num = minNum + 1; num < maxNum; num++)
        {
            if (!nums.Contains(num))
            {
                missingNums.Add(num);
            }
        }

        return missingNums;
    }

    static void Main(string[] args)
    {
        int listLength = int.Parse(Console.ReadLine());
        List<int> nums = new List<int>();

        for (int i = 0; i < listLength; i++)
        {
            int num = int.Parse(Console.ReadLine());
            nums.Add(num);
        }

        List<int> missingNumbers = GetMissingNumbersInSequence(nums);
        string missingNumbersStr = string.Join(" ", missingNumbers);
        Console.WriteLine(missingNumbersStr);
    }
}

File: code-coaches/medium/missing-numbers/MissingNumbers.swift
func getMissingNumbersInSequence(nums: [Int]) -> [Int] {
    let minNum = nums.min() ?? 0
    let maxNum = nums.max() ?? 0
    var missingNums = [Int]()

    for num in (minNum + 1)..<maxNum {
        if !nums.contains(num) {
            missingNums.append(num)
        }
    }

    return missingNums
}

func main() {
    let listLength = Int(readLine()!)!
	var nums = [Int]()

	for _ in 0..<listLength {
		let num = Int(readLine()!)!
		nums.append(num)
	}

	let missingNumbers = getMissingNumbersInSequence(nums: nums)
	let missingNumbersStr = missingNumbers.map(String.init).joined(separator: " ")
	print(missingNumbersStr)
}

main()
